include/micm/solver/cuda_lu_decomposition.hpp

```diff
@@ -0,0 +1,144 @@
+#pragma once
+#include<micm/solver/lu_decomposition.hpp>
+#include<micm/util/cuda_param.hpp>
+#include <stdexcept>
+#ifdef USE_CUDA
+#include <micm/solver/cuda_lu_decomposition.cuh>
+#endif 
+
+#ifdef USE_CUDA
+namespace micm{
+    class CUDALuDecomposition: public LuDecomposition{
+    public: 
+    CUDALuDecomposition(){}; 
+    
+    /// @brief Construct an LU decomposition algorithm for a given sparse matrix
+    /// @param matrix Sparse matrix
+    template<typename T, typename OrderingPolicy>
+    CUDALuDecomposition(const SparseMatrix<T, OrderingPolicy>& matrix); 
```

Comment:
Don't you still need to call the parent constructor here?

Reply:
No. I believe it is called when creating a instance of this class. This occurs at the testing function.

Reply:
I don't think that's true. I think the only constructor called by default is the one with no arguments. Since we have arguments I believe you need to call it explicitly. You can check if this is true with a print statement

Reply:
Yes you are right. I meant this constructor is explicitly called when creating an instance of the class in the test function:  micm::LuDecomposition gpu_lud(A);  Does it need to be call inside the class definition, too?